src/cache/mod.rs

```diff
@@ -5,6 +5,27 @@ lazy_static! {
     static ref GLOBAL_CACHE: Cache = Cache::default();
 }
 
+fn get_dir(uri: String) -> String {
+    let mut parts = uri.split('/').collect::<Vec<&str>>();
+    parts.pop();
+    parts.join("/")
```

Comment:
This could build a lot memory allocation. Should use [rfind](https://doc.rust-lang.org/stable/std/primitive.str.html#method.rfind) , to get the last appearance of the string, and use substring to get the result.

Reply:
I think this implementation is fine - the memory allocation is small, this is a pretty common pattern for parsing directories, and it favors readability.